src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -47,36 +63,40 @@ public static void main(String[] args) {
 
   /**
    * Register the framework with the executor.
-   **/
+   */
   @Override
   public void registered(ExecutorDriver driver, ExecutorInfo executorInfo,
       FrameworkInfo frameworkInfo, SlaveInfo slaveInfo) {
     // Set up data dir
     setUpDataDir();
     createSymbolicLink();
-    log.info("Executor registered with the slave");
+    LOG.info("Executor registered with the slave");
   }
 
   /**
    * Delete and recreate the data directory.
-   **/
+   */
   private void setUpDataDir() {
     // Create primary data dir if it does not exist
-    File dataDir = new File(schedulerConf.getDataDir());
-    if (!dataDir.exists()) {
-      dataDir.mkdirs();
-    }
+    File dataDir = new File(hdfsConfig.getDataDir());
+    createDir(dataDir);
 
     // Create secondary data dir if it does not exist
-    File secondaryDataDir = new File(schedulerConf.getSecondaryDataDir());
-    if (!secondaryDataDir.exists()) {
-      secondaryDataDir.mkdirs();
+    File secondaryDataDir = new File(hdfsConfig.getSecondaryDataDir());
+    createDir(secondaryDataDir);
+  }
+
+  private void createDir(File dataDir) {
+    if (dataDir.exists()) {
+      LOG.warn("data dir exits:" + dataDir);
+    } else if (!dataDir.mkdirs()) {
+      LOG.error("unable to create dir: " + dataDir);
```

Comment:
Return an error/exception? The callers (setUpDataDir, createSymbolicLink) need to know if the createDir was successful.